frontend/src/diagram/DiagramWebSocketContainer.tsx

```diff
@@ -867,10 +867,34 @@ export const DiagramWebSocketContainer = ({
     if (!invokeNodeToolLoading) {
       resetTools();
     }
-  }, [invokeNodeToolLoading, invokeNodeToolData, invokeNodeToolError, handleError, resetTools]);
+    if (invokeNodeToolData) {
+      const { invokeNodeToolOnDiagram } = invokeNodeToolData;
+      const newSelection = invokeNodeToolOnDiagram?.newSelection;
```

Comment:
`invokeNodeToolOnDiagram` cannot be `null` unless I am mistaken, same for `invokeEdgeToolOnDiagram`. In a similar fashion, `entries` cannot be `null` so no need for `?.` The typing should indicate that.

Reply:
I have added the necessary TypeScript typing for all the code manipulated in this PR.